### PR TITLE
http headers are case insensitive

### PR DESCRIPTION
**Pull request by jhunsaker**

### CODE REVIEW

File: pc/net_socket.cpp [hunk 1]
```diff
@@ -1029,16 +1031,19 @@ bool http_server::parse( const char *ptr, size_t len, size_t& res )
     if ( !find( CR, ptr, end ) )  return false;
     assert( hdr_end >= hdr );
     const size_t hlen = static_cast< size_t >( hdr_end - hdr );
+    std::string hdr_str{ hdr, hlen };
```
Comment (tony-ricciardi) [inline pc/net_socket.cpp:1034]: `const`

File: pc/net_socket.cpp [hunk 2]
```diff
@@ -1029,16 +1031,19 @@ bool http_server::parse( const char *ptr, size_t len, size_t& res )
...
+    for ( char& c : hdr_str ) {
+      c = std::toupper( static_cast< unsigned char>( c ) );
```
Comment (tony-ricciardi) [inline pc/net_socket.cpp:1036]: whitespace - should be `< unsigned char >`

**Review by tony-ricciardi — APPROVED** [review]

File: pc/net_socket.cpp (hunk 1 above)
Comment (jhunsaker) [reply pc/net_socket.cpp:1034]: it's modified

**Review by tony-ricciardi — APPROVED** [review]

**Comment by jhunsaker** [comment]
this is a partial fix for issue #113 - more header logic should be made case-insensitive

**Pull request merged**